test/suites/smoke-test-dancebox/test-staking-requests.ts

```diff
@@ -0,0 +1,153 @@
+import { beforeAll, describeSuite, expect } from "@moonwall/cli";
+import { getBlockArray } from "@moonwall/util";
+import type { ApiPromise } from "@polkadot/api";
+import type { GenericExtrinsic } from "@polkadot/types";
+import type { FrameSystemEventRecord } from "@polkadot/types/lookup";
+import type { AnyTuple } from "@polkadot/types/types";
+import Bottleneck from "bottleneck";
+
+const timePeriod = process.env.TIME_PERIOD ? Number(process.env.TIME_PERIOD) : 1 * 60 * 60 * 1000;
+const timeout = Math.max(Math.floor(timePeriod / 12), 5000);
+const hours = (timePeriod / (1000 * 60 * 60)).toFixed(2);
+
+type BlockFilteredRecord = {
+    blockNum: number;
+    extrinsics: GenericExtrinsic<AnyTuple>[];
+    events: FrameSystemEventRecord[];
+    session;
+    pendingOperations;
+};
+
+describeSuite({
+    id: "SM06",
+    title: `Pending Operations in the last ${hours} should be correctly executed`,
+    foundationMethods: "read_only",
+    testCases: ({ it, context, log }) => {
+        let api: ApiPromise;
+        let blockData: BlockFilteredRecord[];
+
+        beforeAll(async () => {
+            api = context.polkadotJs();
+            const blockNumArray = await getBlockArray(api, timePeriod);
+            log(`Collecting ${hours} hours worth of blocks`);
+
+            const getBlockData = async (blockNum: number) => {
```

Comment:
This looks like very similar to the previous file, can we extract it to the utils and re-use in both places?

Reply:
yes, I was thinking about putting it in common, and if the runtime is flashbox, simply skip it